### PR TITLE
Enhanc/Test independence of EMB

### PR DESCRIPTION
**Pull request by JulStraus**
So far, we had a hen and egg problem regarding the tests of `EnergyModelsInvestments`. Since we switched dependencies (EMB having EMX as dependency), we observed the problem that tests being dependent on EMB are not workable if:

* External dependencies are changed, *i.e.*, breaking updates in `Timestruct` or
* Breaking changes in `EnergyModelsInvestments`.

This is solved in this PR using a smaller test model. In addition, the tests have been reworked for understanding potential errors in all functions.

Due to the rewriting of the tests, I identified an error in `RollingLife` when the lifetime was shorter than the duration of the strategic period. This bug was fixed.

### CODE REVIEW

File: test/test_lifetime.jl [hunk 1]
```diff
@@ -1,80 +1,285 @@
...
+    demand = StrategicProfile([10,10,30,35])
+    m, para = simple_model(;inv_data,demand)
+
+    # Extraction of required data
+    n = para[:node]
+    𝒯 = para[:T]
+    𝒯ᴵⁿᵛ = strat_periods(𝒯)
+    inv_data = para[:inv_data]
+    disc_rates = [objective_weight(t_inv, Discounter(para[:disc_rate], 𝒯)) for t_inv ∈ 𝒯ᴵⁿᵛ]
+    capex = StrategicProfile([
```
Comment (hellemo) [inline test/test_lifetime.jl:51]: Perhaps explain the numbers and calculation of capex, not sure I follow why the updates are different, e.g. 15 is not updated.

**Review by hellemo — APPROVED** [review]
Very nice!

Might want to comment/explain where I didn't quite follow.

File: test/test_lifetime.jl [hunk 2]
```diff
@@ -1,80 +1,285 @@
...
+    n = para[:node]
+    𝒯 = para[:T]
+    𝒯ᴵⁿᵛ = strat_periods(𝒯)
+    inv_data = para[:inv_data]
+    capex = StrategicProfile([1,1,1,0])*1e4
+
+    # Tests of the lifetime calculation
+    # - set_capacity_cost(m, element, inv_data, prefix, 𝒯ᴵⁿᵛ, disc_rate, ::UnlimitedLife)
+    @testset "Lifetime calculations" begin
+        # Test that `:cap_rem` is fixed to 0
```
[the remark this answers is not in the document]
Comment (JulStraus) [reply test/test_lifetime.jl:16]: I guess that would be implying we are testing things from JuMP. I am not the biggest fan of introducing tests that test if stuff from other packages are working. In this case, I would think we should rely on JuMP?

File: test/test_lifetime.jl [hunk 3]
```diff
@@ -1,80 +1,285 @@
...
+
+    # Extraction of required data
+    n = para[:node]
+    𝒯 = para[:T]
+    𝒯ᴵⁿᵛ = strat_periods(𝒯)
+    inv_data = para[:inv_data]
+    disc_rate = 1/(1+para[:disc_rate])^5
+    invest = StrategicProfile([5, 10, 15, 15])
+    capex = invest * (1 + disc_rate) * 1e3
+    [capex[t_inv] for t_inv ∈ 𝒯ᴵⁿᵛ]'
```
[the remark this answers is not in the document]
Comment (JulStraus) [reply test/test_lifetime.jl:143]: Yes. That seems to me as I copied things in which i did not remove...

**Comment by JulStraus** [comment]
I included all comments in 299fd9bbac31c25e32c9394f5aa48beccc0d21d2. I found another typo which was also updated.

**Review by hellemo — APPROVED** [review]

File: test/test_lifetime.jl (hunk 2 above)
[the remark this answers is not in the document]
Comment (hellemo) [reply test/test_lifetime.jl:16]: No problem, I was just thinking we test that it is fixed, but not that it's fixed to 0.

**Pull request merged**